resize.js

```diff
@@ -147,9 +147,30 @@ image.get = function (req, res) {
     return;
   }
 
-  if (params.resolutionX < 0 || params.resolutionY < 0 || params.resolutionX > config.get('constraints.max_width') || params.resolutionY > config.get('constraints.max_height')) {
-    res.writeHead(400, 'Bad request');
-    res.end('The requested image size falls outside of the allowed boundaries of this service.');
+  var valid = true;
+  if (params.resolutionX <= 0) {
+     params.resolutionX = 1;
+     valid = false;
+  }
+  if (params.resolutionY <= 0) {
+     params.resolutionY = 1;
+     valid = false;
+  }
```

Comment:
When one of the resolutions is non-positive then I think it's more sensible to error out, because that is clearly a request error. Responding with a 1x1 image doesn't make any sense IMHO.